File: packages/backend/src/main/kotlin/quizzem/backend/features/question/model/AnswerModel.kt
package quizzem.backend.features.question.model

import jakarta.persistence.*
import quizzem.backend.core.domain.QuizzemModel

@Entity
@Table(name = "answers")
class AnswerModel(
    @Column(columnDefinition = "TEXT")
    var text: String,

    var isCorrectAnswer: Boolean = false,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "question_id", nullable = false)
    var question: QuestionModel
) : QuizzemModel()

File: packages/backend/src/features/question/model/answer.model.ts
import { QuizzemModel } from 'src/core/domain/quizzem.model';
import { QuestionModel } from 'src/features/question/model/question.model';
import { Column, Entity, JoinColumn, ManyToOne } from 'typeorm';

@Entity({ name: 'answers' })
export class AnswerModel extends QuizzemModel {
  @Column({ type: 'text' })
  text!: string;

  @Column({ name: 'is_correct_answer', type: 'boolean', default: false })
  isCorrectAnswer: boolean = false;

  @ManyToOne(() => QuestionModel, (question) => question.answers, {
    lazy: true,
    nullable: false,
  })
  @JoinColumn({ name: 'question_id' })
  question!: QuestionModel;
}
